src/main/java/com/alkemy/ong/service/impl/OrganizationServiceImpl.java

```diff
@@ -1,21 +1,51 @@
 package com.alkemy.ong.service.impl;
 
 import com.alkemy.ong.dto.OrganizationDto;
+import com.alkemy.ong.model.Organization;
+import com.alkemy.ong.repository.OrganizationRepository;
 import com.alkemy.ong.service.IOrganizationService;
+
 import com.fasterxml.jackson.databind.ObjectMapper;
+
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
+import org.springframework.web.server.ResponseStatusException;
 
 @Service
 public class OrganizationServiceImpl  implements IOrganizationService {
 
     @Autowired
     ObjectMapper objectMapper;
 
+    @Autowired
+    OrganizationRepository organizationRepository;
+
     @Override
     public OrganizationDto findById(Long id) {
         OrganizationDto organizationFound = objectMapper.convertValue(OrganizationRepository.findById(id), OrganizationDto.class);
         return organizationFound;
     }
 
+    @Override
+    public OrganizationDto updateOrganization(OrganizationDto organizationDto) {
+
+        if(!organizationRepository.existsById(organizationDto.getId())){
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST,
```

Comment:
Aca no seria un BAD REQUEST ya que en realidad la peticion se ejecuto de forma correcta, lo que sucedio es que no encontro ningun registro para el id proporcionado, por lo que seria un NOT FOUND.

Para los mensajes que van hacia la capa de controllers, o sea lo que va a devolver nuestro servicio podriamos implementar un handler y un properties, para no hardcodear los mismos. Te dejo aca unos links para que vayas chusmeando, no hace falta que se implemente ya pero estaria bueno, lo voy a pasar al slack tambien.

https://zetcode.com/spring/messagesource/
https://looksok.wordpress.com/tag/message-properties/